dali/kernels/slice/slice_flip_normalize_permute_common.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SLICE_SLICE_FLIP_NORMALIZE_PERMUTE_COMMON_H_
+#define DALI_KERNELS_SLICE_SLICE_FLIP_NORMALIZE_PERMUTE_COMMON_H_
+
+#include <utility>
+#include <vector>
+#include "dali/core/common.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/slice/slice_kernel_test.h"
+
+namespace dali {
+namespace kernels {
+
+template <typename OutputType, size_t Dims>
+struct SliceFlipNormalizePermuteArgs {
+  std::array<int64_t, Dims> anchor;
+  std::array<int64_t, Dims> shape;
+
+  bool should_pad = false;
+  std::array<int64_t, Dims> padded_shape;
+
+  bool should_flip = false;
+  std::array<bool, Dims> flip;
```

Comment:
Not 100% sure, but I think this
```suggestion
  std::array<bool, Dims> flip = {};
```
will default-initialize the array and you can get rid of `should_flip`.

Reply:
fixed